account_vat_period_end_statement/views/report_vatperiodendstatement.xml

```diff
@@ -8,274 +8,291 @@
                         t-value="account_vat_amounts('debit', statement.debit_vat_account_line_ids)"
                     />
-                           <t t-foreach="vat_accounts" t-as="account_id">
-                               <tr>
-                                   <td class="text-right">Total debit vat</td>
-                                   <td class="text-right"><span
+                    <t t-foreach="vat_accounts" t-as="account_id">
+                        <tr>
+                            <td class="text-right">Total debit vat</td>
+                            <td class="text-right">
+                                <span
                                     t-esc="formatLang(env, vat_accounts[account_id]['amount'])"
-                                /></td>
-                               </tr>
-                               <!-- sum -->
-                               <t
+                                />
+                            </td>
+                        </tr>
+                        <!-- sum -->
+                        <t
                             t-set="total_statement"
                             t-value="(total_statement + vat_accounts[account_id]['amount'])"
                         />
-                           </t>
+                    </t>
 
-                        <!-- tot credit -->
-                        <t
+                    <!-- tot credit -->
+                    <t
                         t-set="vat_accounts"
                         t-value="account_vat_amounts('credit', statement.credit_vat_account_line_ids)"
                     />
-                           <t t-foreach="vat_accounts" t-as="account_id">
-                               <tr>
-                                   <td class="text-right">Total credit vat</td>
-                                   <td class="text-right"><span
+                    <t t-foreach="vat_accounts" t-as="account_id">
+                        <tr>
+                            <td class="text-right">Total credit vat</td>
+                            <td class="text-right">
+                                <span
                                     t-esc="formatLang(env, vat_accounts[account_id]['amount'])"
-                                /></td>
-                               </tr>
-                               <!-- sum -->
-                               <t
+                                />
+                            </td>
+                        </tr>
+                        <!-- sum -->
+                        <t
                             t-set="total_statement"
                             t-value="(total_statement - vat_accounts[account_id]['amount'])"
                         />
-                           </t>
+                    </t>
+
+                    <!-- ...Other tot debit/credit -->
+                    <t
+                        t-foreach="statement.generic_vat_account_line_ids"
```

Comment:
Quindi ora questi valori vengono aggiunti al totale della dichiarazione invece che al totale da pagare, come mai? Pensavo che il totale della dichiarazione dovesse venire solo dai valori delle imposte, e non vedo nulla su questa parte nella issue https://github.com/OCA/l10n-italy/issues/4074.

Visto che questa modifica cambia il comportamento del modulo quindi non è un refactoring, puoi spostarlo in un commit dedicato?

Reply:
spostato in un commit